src/m365/spo/commands/page/page-set.spec.ts

```diff
@@ -490,6 +490,53 @@ describe(commands.PAGE_SET, () => {
     await command.action(logger, { options: { debug: true, name: 'page.aspx', webUrl: 'https://contoso.sharepoint.com/sites/team-a', title: newPageTitle } });
   });
 
+  it('updates page content', async () => {
+    sinonUtil.restore([request.post]);
+
+    const newContent = [{
+      "controlType": 4,
+      "position": {
+        "layoutIndex": 1,
+        "zoneIndex": 1,
+        "sectionIndex": 1,
+        "sectionFactor": 12,
+        "controlIndex": 1
+      },
+      "addedFromPersistedData": true,
+      "innerHTML": "<p>Text content</p>"
+    }];
+
+    const initialPage = {
+      Title: "article",
+      Id: 1,
+      TopicHeader: "TopicHeader",
+      AuthorByline: "AuthorByline",
+      Description: "Description",
+      BannerImageUrl: {
+        Description: '/_layouts/15/images/sitepagethumbnail.png',
+        Url: `https://contoso.sharepoint.com/_layouts/15/images/sitepagethumbnail.png`
+      },
+      CanvasContent1: "{}",
+      LayoutWebpartsContent: "{}"
+    };
+
+    sinon.stub(request, 'post').callsFake(async (opts) => {
+      if ((opts.url as string).includes(`/_api/sitepages/pages/GetByUrl('sitepages/page.aspx')/checkoutpage`)) {
+        return initialPage;
+      }
+      if ((opts.url as string).includes(`/_api/SitePages/Pages(1)/SavePage`) ||
+        (opts.url as string).includes(`/_api/SitePages/Pages(1)/SavePageAsDraft`)) {
+        // Check that the content is being sent correctly
+        assert.deepStrictEqual(JSON.parse(opts.data.CanvasContent1), newContent);
```

Comment:
Instead of using a stub, let's use a spy instead. You can assert on the spy that the correct object is posted.